crates/iroha_core/src/state.rs

```diff
@@ -576,20 +576,10 @@ pub trait WorldReadOnly {
     fn account_permissions_iter<'slf>(
         &'slf self,
         account_id: &AccountId,
-    ) -> Result<std::collections::btree_set::IntoIter<&'slf Permission>, FindError> {
+    ) -> Result<std::collections::btree_set::Iter<'slf, Permission>, FindError> {
         self.account(account_id)?;
 
-        let mut tokens = self
-            .account_inherent_permissions(account_id)
-            .collect::<BTreeSet<_>>();
-
-        for role_id in self.account_roles_iter(account_id) {
-            if let Some(role) = self.roles().get(role_id) {
-                tokens.extend(role.permissions.iter());
-            }
-        }
-
-        Ok(tokens.into_iter())
```

Comment:
I decided to remove this and have `FindPermissions` return only inherent permissions because I was hitting a bug where executor would query all permissions of an account and would then try to revoke them but some of the returned permissions were inside roles and not inherent so the revoke instruction would fail. This was happening with `is_permission_domain_associated` and the likes

Alternatively, I could have taken an approach where this query returns inherent and role permissions but clearly separated so that the caller can know how to work with each and only revoke the inherent ones.